src/batch/src/executor/insert.rs

```diff
@@ -145,12 +142,12 @@ impl InsertExecutor {
                 yield data_chunk.clone();
             }
             for chunk in builder.append_chunk(data_chunk) {
-                write_chunk(chunk)?;
+                notifiers.push(write_chunk(chunk).await?);
             }
         }
 
         if let Some(chunk) = builder.consume_all() {
-            write_chunk(chunk)?;
+            notifiers.push(write_chunk(chunk).await?);
         }
 
         // Wait for all chunks to be taken / written.
```

Comment:
BTW, this is mainly for the linearization of `FLUSH` after `INSERT`, but might be too heavy as we need to wait for the notification for each inserted chunk. May need redesign later.